internal/grpc/grpc.go

```diff
@@ -241,7 +244,7 @@ type RDS struct {
 // Resources returns the contents of RDS"s cache as a []*types.Any.
 // TODO(dfc) cache the results of Resources in the VirtualHostCache so
 // we can avoid the error handling.
-func (r *RDS) Resources() ([]*types.Any, error) {
```

Comment:
you should not have to change the signature of this function.

Reply:
@davecheney I have sticked to not changing the signature as per your suggestion in [commit](https://github.com/heptio/contour/pull/226/commits/f0456317e957b5332cdb792782e4b4ea678cfe21). To achieve this, I had to have a `for` loop to convert the `[]*type.Any` to `[]type.Any` type.

If this is the acceptable approach, I can do the same for the entire PR. I will hold on to get a go ahead from you so that I am sure that I am not making some naive mistake of going down a wrong path.